win32ss/drivers/miniport/xboxvmp/xboxi2c.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * PROJECT:     ReactOS Xbox miniport video driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     I2C SMBus routines
+ * COPYRIGHT:   Copyright 2004 Gé van Geldorp
+ *              Copyright 2004 Filip Navara
+ *              Copyright 2019 Stanislav Motylkov (x86corez@gmail.com)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "xboxvmp.h"
+
+#include <debug.h>
+#include <dpfilter.h>
+
+/* PUBLIC AND PRIVATE FUNCTIONS ***********************************************/
+
+static
+BOOLEAN
+ReadfromSMBus(
+    UCHAR Address,
+    UCHAR bRegister,
+    UCHAR Size,
+    ULONG *Data_to_smbus)
+{
+    int nRetriesToLive = 50;
+
+    while ((VideoPortReadPortUshort((PUSHORT) (I2C_IO_BASE + 0)) & 0x0800) != 0)
+    {
+        ; /* Franz's spin while bus busy with any master traffic */
+    }
+
+    while (nRetriesToLive-- != 0)
+    {
+        UCHAR b;
+        int temp;
```

Comment:
more descriptive?

Reply:
I just moved this code from one location to another. :sweat_smile: I'll probably refactor it later.